substrate/client/network/sync/src/strategy/chain_sync.rs

```diff
@@ -1330,16 +1369,11 @@ where
 			}
 
 			// handle peers that were in other states.
-			let action = match self.add_peer_inner(peer_id, p.best_hash, p.best_number) {
-				// since the request is not a justification, remove it from pending responses
-				Ok(None) => ChainSyncAction::CancelBlockRequest { peer_id },
-				// update the request if the new one is available
-				Ok(Some(request)) => ChainSyncAction::SendBlockRequest { peer_id, request },
-				// this implies that we need to drop pending response from the peer
-				Err(bad_peer) => ChainSyncAction::DropPeer(bad_peer),
-			};
+			self.add_peer(peer_id, p.best_hash, p.best_number);
 
-			self.actions.push(action);
+			// since the request is not a justification, remove it from pending responses
+			// TODO: check `PeerSyncState`?
```

Comment:
?

Reply:
Not every `PeerSyncState` state has an associated pending response. On the other hand, removing pending response when there is actually no active one is harmless, so to not complicate the code I think we don't have to check `PeerSyncState`.